astropy/coordinates/transformations.py

```diff
@@ -347,23 +347,34 @@ def get_transform(self, fromsys, tosys):
             self._composite_cache[fttuple] = comptrans
         return self._composite_cache[fttuple]
 
-    def lookup_name(self, name):
+    def lookup_name(self, name, raises=False):
         """
         Tries to locate the coordinate class with the provided alias.
 
         Parameters
         ----------
         name : str
             The alias to look up.
-
+        raises : Bool (default value is False)
+            False : doesn't raise exception if 'name' isn't found. (returns None)
+            True : raises exception if 'name' isn't found.
         Returns
         -------
         coordcls
+        if raises == False:
             The coordinate class corresponding to the ``name`` or `None` if
             no such class exists.
+        if raises == True:
+            The coordinate class corresponding to the ``name`` or raises Exception
+            if no such class exists.
         """
-
-        return self._cached_names.get(name, None)
+        if not raises:
+            return self._cached_names.get(name, None)
+        else:
+            if self._cached_names.get(name, None) is None:
+                raise Exception("name not found")
+            else:
+                return self._cached_names.get(name, None)
```

Comment:
It might be cleaner to simply do:

```
if name in self._cached_names:
    return return self._cached_names[name]
elif raises:
    raise Exception(...)
```

I think we might also want a better keyword name than ``raises``, but I'll let the astropy.coordinates maintainers comment on this.

Reply:
@astrofrog Sure, I'll make the above mentioned changes. Also, can you please tell me which test file should be amended?

Reply:
The test would need to be added here: https://github.com/astropy/astropy/blob/master/astropy/coordinates/tests/test_transformations.py

In general if you are looking at file ``astropy/package/blah.py``, there should usually be a matching ``astropy/package/tests/test_blah.py``.

Reply:
@astrofrog  Thanks! I'll add the tests. Also, can you please tell me why is the travis test failing? The reason seems unclear.

Reply:
The failure is unrelated and just a web-resource that is temporarily unavailable. When you push the test(s) it will rebuild and (probably) pass.